core/modules/filter/filter.module

```diff
@@ -133,6 +133,12 @@ function filter_menu() {
     'access arguments' => array(4),
     'file' => 'filter.admin.inc',
   );
+  $items['filter-link/autocomplete'] = array(
```

Comment:
At first I was not sure about this path, but I see it matches `node/autocomplete` and `user/autocomplete` etc. So it seems fine to me.